From 37e50a2a35cadcb5318292fecaca5b1bf9b457f7 Mon, 10 Oct 2016 15:49:25 +0200
From: Thomas Perl <thomas.perl@jollamobile.com>
Date: Tue, 13 Aug 2013 13:36:33 +0200
Subject: [PATCH] Force repo refresh in pk_backend_refresh_cache


This is in line with what most other backends do: When a refresh
of the cache is requested, we want to force a cache update. The
command-line "pkcon" client doesn't provide a way to set force.

[backends] zypp: Force repository refresh

diff --git a/backends/zypp/pk-backend-zypp.cpp b/backends/zypp/pk-backend-zypp.cpp
index 275469c..d5ab8e3 100644
--- a/backends/zypp/pk-backend-zypp.cpp
+++ b/backends/zypp/pk-backend-zypp.cpp
@@ -2176,11 +2176,6 @@
 static void
 backend_refresh_cache_thread (PkBackendJob *job, GVariant *params, gpointer user_data)
 {
-	gboolean force;
-	g_variant_get (params, "(b)",
-		       &force);
-
-	MIL << force << endl;
 	ZyppJob zjob(job);
 	ZYpp::Ptr zypp = zjob.get_zypp();
 
@@ -2189,7 +2184,7 @@
 		return;
 	}
 
-	zypp_refresh_cache (job, zypp, force);
+	zypp_refresh_cache (job, zypp, TRUE);
 	pk_backend_job_finished (job);
 }
 
